components/sentry/sentry.app.mjs

```diff
@@ -0,0 +1,299 @@
+    },
+    async _paginate(maxResults = 1000, apiRequestFunction, ...params) {
+      let data = [];
+      let cursor = null;
+      do {
+        // Here we make the request using the params and the cursor
+        const res = await apiRequestFunction(...params, cursor?.cursor);
+        cursor = parseLinkHeader(res.headers["link"]).next;
+
+        // Add the results to the data array
+        data.push(...res.data);
+
+        // If we reach the maxResults limit, break the loop and slice the array
+        if (data.length > maxResults) {
+          data = data.slice(0, maxResults);
+          break;
+        }
+
+        // Break the loop if there are no more results
+        if (cursor.results == "false") {
+          break;
+        }
+      } while (true);
+      return data;
+    },
+    _organizationObjectToOption(organization) {
+      const {
+        name,
+        slug,
+      } = organization;
+      const label = `${name} (${slug})`;
+      return {
+        label,
+        value: slug,
+      };
+    },
+    async _propDefinitionsOptions(
+      url,
+      params, {
+        page,
+        prevContext,
+      },
+    ) {
+      let requestConfig = this._makeRequestConfig();  // Basic axios request config
+      if (page === 0) {
+        // First time the options are being retrieved.
+        // Include the parameters provided, which will be persisted
+        // across the different pages.
+        requestConfig = {
+          ...requestConfig,
+          params,
+        };
+      } else if (prevContext.nextPage) {
+        // Retrieve next page of options.
+        url = prevContext.nextPage.url;
+      } else {
+        // No more options available.
+        return {
+          data: [],
+        };
+      }
+
+      const {
+        data,
+        headers: { link },
+      } = await axios.get(url, requestConfig);
+      // https://docs.sentry.io/api/pagination/
+      const { next } = parseLinkHeader(link);
+
+      return {
+        data,
+        next,
+      };
+    },
+    _baseIntegrationParams() {
+      return {
+        scopes: [
+          "event:read",
+        ],
+        events: [
+          "issue",
+        ],
+        isAlertable: true,
+        isInternal: true,
+        verifyInstall: false,
+      };
+    },
+    _formatIntegrationName(rawName) {
+      const options = {
+        remove: /[()]/g,
+        lower: true,
+      };
+      const enrichedRawName = `pd-${rawName}`;
+      return slugify(enrichedRawName, options).substring(0, 57);
+    },
+    async createIntegration(eventSourceName, organization, webhookUrl) {
+      const url = this._integrationsEndpoint();
+      const name = this._formatIntegrationName(eventSourceName);
+      const requestData = {
+        ...this._baseIntegrationParams(),
+        name,
+        organization,
+        webhookUrl,
+      };
+      const requestConfig = this._makeRequestConfig();
+      const { data } = await axios.post(url, requestData, requestConfig);
```

Comment:
Optional:

Don't you think it would be better to have a `makeRequest` function unifying the `axios` lib use there using the `config` object as the only parameter?
something like
```javascript
return axios({...});
```

Reply:
Humm. It may be a good idea, but I guess the original developer did this way because of some specifications about sentry pagination. They use a `link` header which makes things a little bit more complicated to paginate. I would prefer maintain this way :)